### PR TITLE
[web] fix leaking problem in tests for text editing.

### PR DESCRIPTION
**Pull request by nturgut**
fix leaking problem in tests for text editing. removing skips for firefox

related to: https://github.com/flutter/flutter/issues/46638

### CODE REVIEW

File: lib/web_ui/test/text_editing_test.dart [hunk 1]
```diff
@@ -975,11 +976,19 @@ void main() {
...
+        textarea.dispatchEvent(Event.eventType('Event', 'input'));
+        textarea.setSelectionRange(2, 5);
+        Event keyup = KeyboardEvent('keyup');
+        textEditing.editingElement.domElement.dispatchEvent(keyup);
+        document.dispatchEvent(Event.eventType('Event', 'selectionchange'));
+      } else {
+        textarea.dispatchEvent(Event.eventType('Event', 'input'));
+        textarea.setSelectionRange(2, 5);
+        document.dispatchEvent(Event.eventType('Event', 'selectionchange'));
+      }
```
Comment (yjbanov) [inline lib/web_ui/test/text_editing_test.dart]: This part can be deduplicated a bit:

```dart
      textarea.dispatchEvent(Event.eventType('Event', 'input'));
      textarea.setSelectionRange(2, 5);
      if (browserEngine == BrowserEngine.firefox) {
        textEditing.editingElement.domElement.dispatchEvent(KeyboardEvent('keyup'));
      }
      document.dispatchEvent(Event.eventType('Event', 'selectionchange'));
```

File: lib/web_ui/test/text_editing_test.dart [hunk 2]
```diff
@@ -1270,6 +1272,18 @@ void checkInputEditingState(
...
 }
 
+/// In case of an exception backup DOM element(s) can still stay on the DOM.
+void clearBackUpDomElementIfExists() {
+  List<Node> domElementsToRemove = List<Node>();
+  if (document.getElementsByTagName('input').length > 0) {
+    domElementsToRemove..addAll(document.getElementsByTagName('input'));
+  }
+  if (document.getElementsByTagName('textarea').length > 0) {
+    domElementsToRemove..addAll(document.getElementsByTagName('input'));
```
Comment (yjbanov) [inline lib/web_ui/test/text_editing_test.dart]: Did you mean to get 'textarea'?

File: lib/web_ui/test/text_editing_test.dart (hunk 1 above)
Comment (nturgut) [reply lib/web_ui/test/text_editing_test.dart]: done. thanks!

File: lib/web_ui/test/text_editing_test.dart (hunk 2 above)
Comment (nturgut) [reply lib/web_ui/test/text_editing_test.dart]: done

**Pull request merged**